### PR TITLE
Config redesign

### LINKED ISSUE #213: Recipes: Removal
### Cookbook version
4.0.0-beta

### Chef-client version
Latest

### Platform Details
- All supported

### Scenario:
As a user
I do not want to be using any recipes or attributes, all required properties should be using custom resources

### Expected Result:
- [ ] grafana is configurable via resource properties
- [ ] grafana has unit tests 
- [ ] there are no more recipes in grafana
- [ ] there are no more attributes in grafana

### PR DESCRIPTION
**Pull request by xorima**
### Description

Enables us to manage the configuration file using resources

### Issues Resolved

#213 

### Check List
- [ ] All tests pass. See https://github.com/sous-chefs/grafana/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by JohnRoesler — DISMISSED** [review]
Huge PR! Looking good to me, pending the failing circle tests

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,25 +14,25 @@ workflows:
       - kitchen/dokken-single:
           name: centos-7
           platform: centos-7
-          suite: default
+          suite: install
```
Comment (damacus) [inline .circleci/config.yml]: If this is the default set of tests, we can probably just use default for simplicity.

"Hey this is the default setup, and it's mostly covered in this suite"

File: Berksfile [hunk 2]
```diff
@@ -1,7 +1,9 @@
 source 'https://supermarket.chef.io'
 
 metadata
+
 group :integration do
+  cookbook 'test', path: 'test/fixtures/cookbooks/test'
   cookbook 'apt'
   cookbook 'curl'
```
Comment (damacus) [inline Berksfile]: These two can get removed, they should be in the metadata of the cookbook not here :)

File: resources/config.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+property  :group,               String, default: 'grafana'
+property  :restart_on_upgrade,  String, default: 'false'
+property  :conf_directory,      String, default: '/etc/grafana'
+property  :config_file,         String, default: lazy { ::File.join(conf_directory, 'grafana.ini') }
+property  :app_mode,            String, default: 'production', equal_to: %w(production development)
+property  :cookbook,            String, default: 'grafana'
+property  :source,              String, default: 'grafana.ini.erb'
+
+action :install do
+  user new_resource.owner do
```
Comment (damacus) [inline resources/config.rb]: any resource 


without config in between the do end blocks can get rid of the do end blocks. 
e.g. 
```suggestion
  user new_resource.owner
```

File: spec/unit/default_spec.rb [hunk 4]
```diff
@@ -1,240 +1,23 @@
...
-            expect(chef_run).to render_file('/etc/grafana/ldap.toml').with_content(/^\[\[servers.group_mappings\]\]/)
-            expect(chef_run).to render_file('/etc/grafana/ldap.toml').with_content(/^group_dn = "*"/)
-            expect(chef_run).to render_file('/etc/grafana/ldap.toml').with_content(/^org_role = "Viewer"/)
-          end
-
-          it 'enable grafana-server service' do
-            expect(chef_run).to enable_service('grafana-server')
-          end
-        end
+platforms = {
```
Comment (damacus) [inline spec/unit/default_spec.rb]: For a better time testing custom resources look at https://github.com/sous-chefs/apache2/tree/master/spec/resources

**Review by damacus — CHANGES_REQUESTED** [review]

File: test/fixtures/cookbooks/test/recipes/install.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
+grafana_install 'grafana' do
+end
+
+grafana_config 'Grafana' do
```
Comment (damacus) [inline test/fixtures/cookbooks/test/recipes/install.rb]: Fix me please

**Review by damacus — APPROVED** [review]

**Pull request merged**